example-app/src/main/java/com/adyen/checkout/example/ui/bacs/BacsFragment.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 3/1/2023.
+ */
+
+package com.adyen.checkout.example.ui.bacs
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.adyen.checkout.bacs.BacsDirectDebitComponent
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.example.databinding.FragmentBacsBinding
+import com.adyen.checkout.example.ui.configuration.CheckoutConfigurationProvider
+import com.adyen.checkout.redirect.RedirectComponent
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class BacsFragment : BottomSheetDialogFragment() {
+
+    @Inject
+    internal lateinit var checkoutConfigurationProvider: CheckoutConfigurationProvider
+
+    private var _binding: FragmentBacsBinding? = null
+    private val binding: FragmentBacsBinding get() = requireNotNull(_binding)
+
+    private val viewModel: BacsViewModel by viewModels()
+
+    private var bacsComponent: BacsDirectDebitComponent? = null
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        // Insert return url in extras, so we can access it in the ViewModel through SavedStateHandle
+        val returnUrl = RedirectComponent.getReturnUrl(requireActivity().applicationContext)
```

Comment:
Should this have `+ "/bacs"`?

Reply:
Not sure, because a fragment can't really handle the return url. I figured the default one would do, so we could potentially handle it in `MainActivity` then. Also, I didn't really implement the return url flow, because I thought BACS doesn't have it (currently)

Reply:
Hmm ok then we'll look at it later with some component that produces actions